src/main/java/org/openrewrite/java/migrate/AddMissingMethodImplementation.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.*;
+import org.openrewrite.internal.lang.NonNull;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeTree;
+
+import java.util.List;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class AddMissingMethodImplementation extends Recipe {
+    @Option(displayName = "Fully Qualified Class Name",
+            description = "A fully qualified class being implemented with missing method.",
+            example = "com.yourorg.FooBar")
+    @NonNull
+    String fullyQualifiedClassName;
+
+    @Option(displayName = "Method Pattern",
+            description = "A method pattern for matching required method definition.",
+            example = "*..* hello(..)")
+    @NonNull
+    String methodPattern;
+
+    @Option(displayName = "Method Template",
+            description = "Template of method to add",
+            example = "public String hello() { return \\\"Hello from #{}!\\\"; }")
+    @NonNull
+    String methodTemplateString;
+
+    // All recipes must be serializable. This is verified by RewriteTest.rewriteRun() in your tests.
+    @JsonCreator
+    public AddMissingMethodImplementation(@NonNull @JsonProperty("fullyQualifiedClassName") String fullyQualifiedClassName,
+                                          @NonNull @JsonProperty("methodPattern") String methodPattern,
+                                          @NonNull @JsonProperty("methodTemplateString") String methodTemplateString) {
+        this.fullyQualifiedClassName = fullyQualifiedClassName;
+        this.methodPattern = methodPattern;
+        this.methodTemplateString = methodTemplateString;
+    }
+
```

Comment:
Does the Lombok `@Value` annotation not already provide us with a workable constructor?
```suggestion
```

Reply:
From my experience tests have been failing without the constructor. I can try removing it and see if the tests still work.

Reply:
Are you passing `-parameters` to the compiler? Don't know your IDE or setup, but I'd expect this to work without the explicit constructor.